rust/connlib/tunnel/src/peer.rs

```diff
@@ -378,6 +381,29 @@ impl ClientOnGateway {
                 self.filters.insert(*ip, filter_engine);
             }
         }
+
+        for (addr, TranslationState { resource_id, .. }) in &self.permanent_translations {
+            let Some(resource) = self.resources.get(resource_id) else {
+                continue;
+            };
+
+            debug_assert!(resource.is_dns());
+
+            let mut filter_engine = FilterEngine::empty();
```

Comment:
Is there a way we can de-duplicate this with the above bit?

Reply:
yeah

Reply:
LMK what you think now